src/functionalTest/java/uk/gov/hmcts/reform/locationrefdata/cameltest/LrdBuildingLocationFileStatusCheck.java

```diff
@@ -0,0 +1,176 @@
+package uk.gov.hmcts.reform.locationrefdata.cameltest;
+
+import net.thucydides.core.annotations.WithTag;
+import net.thucydides.core.annotations.WithTags;
+import org.apache.camel.test.spring.junit5.CamelSpringBootTest;
+import org.apache.camel.test.spring.junit5.CamelTestContextBootstrapper;
+import org.apache.camel.test.spring.junit5.MockEndpoints;
+import org.javatuples.Pair;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.JobParametersBuilder;
+import org.springframework.batch.test.JobLauncherTestUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.jpa.JpaRepositoriesAutoConfiguration;
+import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import uk.gov.hmcts.reform.data.ingestion.configuration.AzureBlobConfig;
+import uk.gov.hmcts.reform.data.ingestion.configuration.BlobStorageCredentials;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.LrdIntegrationBaseTest;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.SpringStarter;
+import uk.gov.hmcts.reform.locationrefdata.config.LrdCamelConfig;
+import uk.gov.hmcts.reform.locationrefdata.configuration.BatchConfig;
+
+import java.io.FileInputStream;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.springframework.util.ResourceUtils.getFile;
+import static uk.gov.hmcts.reform.data.ingestion.camel.util.MappingConstants.SCHEDULER_START_TIME;
+
+@TestPropertySource(properties = {"spring.config.location=classpath:application-integration.yml,"
+    + "classpath:application-leaf-integration.yml"})
+@CamelSpringBootTest
+@MockEndpoints("log:*")
+@ContextConfiguration(classes = {LrdCamelConfig.class, CamelTestContextBootstrapper.class,
+    JobLauncherTestUtils.class, BatchConfig.class, AzureBlobConfig.class, BlobStorageCredentials.class},
+    initializers = ConfigFileApplicationContextInitializer.class)
+@SpringBootTest
+@EnableAutoConfiguration(exclude = JpaRepositoriesAutoConfiguration.class)
+@EnableTransactionManagement
+@SqlConfig(dataSource = "dataSource", transactionManager = "txManager",
+    transactionMode = SqlConfig.TransactionMode.ISOLATED)
+@WithTags({@WithTag("testType:Functional")})
```

Comment:
Should not this be `testType:Integration`?

Reply:
I think this is a convention that has to be fixed. There are no integration tests in this repo, only functional tests exist. However, in the resources, we have `application-integration.yaml`. But the `build.gradle` file has functional task defined. So, the tag should be `functional` but the yaml should be renamed to `application-functional.yaml`

Reply:
ah ok. so this is an existing issue which needs to be fixed separately but not as part of this ticket